### PR TITLE
fix: @typescript-eslint incompatible versions with eslint@8.x.x

### PR DESCRIPTION
**Pull request by danielfigueiredo**
## Hey, I just made a Pull Request!

This issue is happening when running `yarn backstage-cli versions:bump`, a full detailed explanation of the error can be found in [this issue](https://github.com/backstage/backstage/issues/27500).

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation N/A
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes) N/A
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by andrewthauer — APPROVED** [review]

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks @danielfigueiredo, left a few comments. Seems other reports of this so let's try to push this forward. 👍

File: packages/cli/package.json [hunk 1]
```diff
@@ -77,8 +77,8 @@
     "@swc/jest": "^0.2.22",
     "@types/jest": "^29.5.11",
     "@types/webpack-env": "^1.15.2",
-    "@typescript-eslint/eslint-plugin": "^6.12.0",
-    "@typescript-eslint/parser": "^6.7.2",
+    "@typescript-eslint/eslint-plugin": "^7.18.0",
+    "@typescript-eslint/parser": "^7.18.0",
```
[the remark this answers is not in the document]
Comment (danielfigueiredo) [reply packages/cli/package.json:81]: I think 8 has more changes required in the tool chain, but I can try and see if anything comes up!

[the remark this answers is not in the document]
Comment (danielfigueiredo) [reply packages/cli/package.json]: Upgraded all eslint and plugins to latest! 🎉

**Comment by danielfigueiredo** [comment]
@awanlin, apologies for the delay, but I've addressed the comments and bumped every lint plugin down to the latest!

**Comment by andrewthauer** [comment]
Any chance you could re-review this @awanlin? Thanks!

**Review by awanlin — APPROVED** [review]
Thanks @danielfigueiredo, looks good to me now. You'll still need the maintainers to review.

**Comment by awanlin** [comment]
Hi @benjdlambert, any chance you can give this one a look, please? Got a few reports of this being an issue and this looks to be something we'd want to do anyway. Now would also be a good time to get it in as we are at the start of the monthly release cycle.

**Review by benjdlambert — APPROVED** [review]

**Comment by danielfigueiredo** [comment]
> Hmm, ok, happy to try this out seen as it didn't seem to break anything in the build. Still a little unsure on the impact to other users, but let's test this internally in the `-next` release too.

@benjdlambert let me know if it causes any undesired side-effect, happy to investigate and patch!

Usually, the plugins and the eslint version need be kept up-to-date together as newer versions can be incompatible with older plugin ones

**Pull request merged**